test/reconnect_test.go

```diff
@@ -722,3 +722,66 @@ func TestReconnectTLSHostNoIP(t *testing.T) {
 		t.Fatalf("ReconnectedCB should have been triggered: %v", nc.LastError())
 	}
 }
+
+func TestConnCloseNoCallback(t *testing.T) {
+	ts := startReconnectServer(t)
+	defer ts.Shutdown()
+
+	cch := make(chan bool)
+	opts := reconnectOpts
+	opts.ClosedCB = func(_ *nats.Conn) {
+		cch <- true
+	}
+	opts.NoCallbacksAfterClientClose = true
+
+	nc, err := opts.Connect()
+	if err != nil {
+		t.Fatalf("Should have connected ok: %v", err)
+	}
+	defer nc.Close()
+
+	nc.Flush()
+	// Close the connection, we don't expect to get a notification
+	nc.Close()
+	// Shutdown the server
+	ts.Shutdown()
+
+	// Even on Windows (where a createConn takes more than a second)
```

Comment:
Waiting 3 secs is too much. A 500ms is more than enough for the callback to be invoked if it was.